host/src/lib/HelperFunctions.jsx

```diff
@@ -0,0 +1,114 @@
+* @returns {array} teamsPickedChoices - array of objects containing corresponding team names and choice texts
+*/
+export const getTeamByQuestion = (teamsArray, currentQuestionIndex, choices, questions, currentState) => {
+  const teamsPickedChoices = [];
+
+  teamsArray.forEach(team => {
+    const teamPickedChoices = [];
+    let isNoResponse = true;
+    
+    team.teamMembers && team.teamMembers.forEach(teamMember => {
+      teamMember.answers && teamMember.answers.forEach(answer => {
+        if (answer.questionId === questions[currentQuestionIndex].id) {
+          if (((currentState === GameSessionState.CHOOSE_CORRECT_ANSWER || currentState === GameSessionState.PHASE_1_DISCUSS) && answer.isChosen) || ((currentState === GameSessionState.PHASE_2_DISCUSS || currentState === GameSessionState.CHOOSE_TRICKIEST_ANSWER) && answer.isTrickAnswer)) {
+            isNoResponse = false;
+            choices && choices.forEach(choice => {
+              if (answer.text === choice.text) {
+                teamPickedChoices.push({
+                  teamName: team.name,
+                  choiceText: choice.text
+                });
+              }
+            });
+          }
+        }
+      });
+    });
+
+    if (teamPickedChoices.length > 0 || isNoResponse) {
+      teamsPickedChoices.push(...teamPickedChoices);
+      if (isNoResponse) {
+        teamsPickedChoices.push({
+          teamName: team.name,
+          choiceText: 'No response'
+        });
+      }
+    }
+  });
+
+  return teamsPickedChoices;
+};
+
+/*
+* returns an array ordered to match the order of answer choices, containing the total number of each answer
+* @param {array} choices - array of choices for current question
+* @param {array} teamsArray - array of teams
+* @param {number} currentQuestionIndex - index of current question
+* @returns {array} answersArray - ordered array of total number of each answer
+*/
+export const getAnswersByQuestion = (choices, teamsArray, currentQuestionIndex, questions, currentState) => {
+  if (teamsArray.length !== 0 && questions && choices && Object.keys(teamsArray[0]).length !== 0 && Object.getPrototypeOf(teamsArray[0]) === Object.prototype) {
+    let choicesTextArray = [choices.length];
+    let answersArray = new Array(choices.length).fill(0);
+    let currentQuestionId = questions[currentQuestionIndex].id;
+    choices && choices.forEach((choice, index) => {
+      choicesTextArray[index] = choice.text;
+    });
+
+    teamsArray.forEach(team => {
+      team.teamMembers && team.teamMembers.forEach(teamMember => {
+        teamMember.answers && teamMember.answers.forEach(answer => {
+          if (answer.questionId === currentQuestionId) {
+            if (((currentState === GameSessionState.CHOOSE_CORRECT_ANSWER || currentState === GameSessionState.PHASE_1_DISCUSS) && answer.isChosen) || ((currentState === GameSessionState.PHASE_2_DISCUSS || currentState === GameSessionState.CHOOSE_TRICKIEST_ANSWER) && answer.isTrickAnswer)) {
```

Comment:
Looks like it's also duplicate as above, worth to extract it into a separate method

Reply:
Nice, this has been done via `extractAnswers`